hibike/hibike_message.py

```diff
@@ -432,6 +434,43 @@ def read(serial_conn):
 
   return HibikeMessage(messageID, payload)
 
+# Sends this message
+# Computes the checksum
+# Then sends each byte of the message, and finally sends the checksum byte
+def hid_send(hid, message):
+
+  """ Sends this message
+      Computes the checksum
+      Then sends each byte of the message, and finally sends the checksum byte
+
+      returns whether or not it sent sucessfully
+  """
+  m_buff = message.toByte()
+  chk = checksum(m_buff)
+  m_buff.append(chk)
+  encoded = cobs_encode(m_buff)
+  out_buf = bytearray([0x00, len(encoded)]) + encoded
+  if (len(out_buf) > MAX_PACKET_SIZE):
+    return False
+  else:
+    hid.write(out_buf)
+    return True
+
+# behold the beauty of sending packets over a packet based communication protocol
+def hid_blocking_receive(hid):
```

Comment:
one reads a single packet and one continually yields packets and never returns. `hibike_process` uses the generator but the other function may be useful for writing tests that want to receive a finite number of packets.